service/start.go

```diff
@@ -99,13 +99,32 @@ func (svc *service) startNostr(ctx context.Context, encryptionKey string) error
 			// register a subscriber for events of "app_created" which handles creation of nostr subscription for new app
 			svc.eventPublisher.RegisterSubscriber(&createAppConsumer{svc: svc, relay: relay})
 
-			// legacy single wallet subscription - only subscribe once for all legacy apps
-			// to ensure we do not get duplicate events
-			err = svc.startAppWalletSubscription(ctx, relay, svc.keys.GetNostrPublicKey(), svc.keys.GetNostrSecretKey())
-			if err != nil {
-				//err being non-nil means that we have an error on the websocket error channel. In this case we just try to reconnect.
-				logger.Logger.WithError(err).Error("Got an error from the relay while listening to subscription.")
-				continue
+			// check if there are still legacy apps in DB
+			var legacyAppCount int64
+			result := svc.db.Model(&db.App{}).Where("wallet_pubkey IS NULL").Count(&legacyAppCount)
+			if result.Error != nil {
+				logger.Logger.WithError(result.Error).Error("Failed to count Legacy Apps")
+				return
+			}
+			if legacyAppCount > 0 {
+				// re-publish single NIP47 event info for legacy apps
+				_, err := svc.GetNip47Service().PublishNip47Info(ctx, relay, svc.keys.GetNostrPublicKey(), svc.keys.GetNostrSecretKey(), svc.lnClient)
+				if err != nil {
+					logger.Logger.WithError(err).Error("Could not publish NIP47 info for legacy apps")
+					continue
+				}
+
+				// legacy single wallet subscription - only subscribe once for all legacy apps
+				// to ensure we do not get duplicate events
+				err = svc.startAppWalletSubscription(ctx, relay, svc.keys.GetNostrPublicKey(), nil)
+				if err != nil {
+					//err being non-nil means that we have an error on the websocket error channel. In this case we just try to reconnect.
+					logger.Logger.WithError(err).Error("Got an error from the relay while listening to subscription.")
+					continue
+				}
+			} else {
```

Comment:
This seems wrong. This is done inside a loop that will re-connect to the relay on errors. But here we only listen for context.Done. I think this will cause a deadlock if the relay disconnects, but no exit signal has happened, right?

Also having the `non-legacy` logic inside an `else` kind of feels wrong, like a code smell to me.

Reply:
replaced with `<-relay.Context().Done()` and handle `relay.ConnectionError` to continue in the loop